src/relay/backend/te_compiler.cc

```diff
@@ -134,41 +144,78 @@ class TECompilerImpl : public TECompilerNode {
     return mod;
   }
 
+  void AddExterns(IRModule module) {
+    // Everything tagged with "Compiler" has been compiled, so remove those definitions.
+    std::vector<GlobalVar> to_be_deleted;
+    for (const auto& kv : module->functions) {
+      if (kv.second->GetAttr<String>(attr::kCompiler).defined()) {
+        to_be_deleted.push_back(kv.first);
+      }
+    }
+    for (const auto& global_var : to_be_deleted) {
+      module->Remove(global_var);
+    }
+    // HOWEVER we still need a Relay definition to go with those now external functions, so
+    // retrieve them from the cache and mark them with "ExternalSymbol".
```

Comment:
Why do you retrieve from the cache instead of using the to_be_deleted list you just constructed?

Reply:
The now-compiled globals have just been removed. This second loop is looking for the copies of those stored away inside the cache entry's 'funcs' IRModule. It was like this when I got here.